mvc/src/main/java/cz/muni/fi/pa165/mushrooms/mvc/controllers/ForestController.java

```diff
@@ -0,0 +1,166 @@
+        return "redirect:" + uriBuilder.path("/forests").build().toUriString();
+    }
+
+
+    @RequestMapping(value = "/edit/{id}", method = RequestMethod.POST)
+    public String update(@PathVariable long id,
+                         @Valid @ModelAttribute("forestEdit") ForestDTO formBean,
+                         BindingResult bindingResult,
+                         Model model,
+                         HttpServletRequest request,
+                         UriComponentsBuilder uriBuilder,
+                         RedirectAttributes redirectAttributes) {
+
+        String res = Tools.redirectNonAdmin(request, uriBuilder, redirectAttributes);
+        if(res != null) return res;
+
+        log.debug("Forest - update");
+        if (bindingResult.hasErrors()) {
+            log.debug("Forest - has errors:");
+
+            for (ObjectError ge : bindingResult.getGlobalErrors()) {
+                log.debug("ObjectError: {}", ge);
+            }
+            for (FieldError fe : bindingResult.getFieldErrors()) {
+                model.addAttribute(fe.getField() + "_error", true);
+                log.debug("FieldError: {}", fe);
+            }
+            model.addAttribute("forestEdit", formBean);
+            return "forests/edit";
+        }
+
+        log.debug("[FOREST] Update: {}", formBean);
+        ForestDTO result = forestFacade.updateForest(formBean);
+
+        redirectAttributes.addFlashAttribute("alert_success", "Forest " + result.getName() + " was updated");
+        return "redirect:" + uriBuilder.path("/forests/read/{id}").buildAndExpand(id).encode().toUriString();
+    }
+
+
+    @RequestMapping(value = "/new", method = RequestMethod.POST)
+    public String create(@Valid @ModelAttribute("forestCreate") ForestDTO formBean,HttpServletRequest request,
+                         BindingResult bindingResult, Model model, RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriBuilder) {
+
+        String res = Tools.redirectNonAdmin(request, uriBuilder, redirectAttributes);
+        if(res != null) return res;
+
+        log.debug("create(forestCreate={})", formBean);
+        //in case of validation error forward back to the the form
+        if (bindingResult.hasErrors()) {
+            for (ObjectError ge : bindingResult.getGlobalErrors()) {
+                log.trace("ObjectError: {}", ge);
+            }
+            for (FieldError fe : bindingResult.getFieldErrors()) {
+                model.addAttribute(fe.getField() + "_error", true);
+                log.trace("FieldError: {}", fe);
+            }
+            return "forests/register";
+        }
+
+        if(forestFacade.findByName(formBean.getName()) != null){
+            redirectAttributes.addFlashAttribute("alert_warning", "Forest " + formBean.getName() + " already exists!");
+            return "redirect:" + uriBuilder.path("/forests").build().toUriString();
```

Comment:
what about  return "redirect:" + uriBuilder.path("/forests/read/{id}").buildAndExpand(formBean.getId()).encode().toUriString();
to show the existing forest ...

Reply:
could be nice, maybe ill have the time